temperature/routers.py

```diff
@@ -0,0 +1,40 @@
+from datetime import datetime
+
+from fastapi import APIRouter, Depends, HTTPException
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from temperature import crud, schemas
+from database import AsyncSessionLocal
+
+router_temp = APIRouter()
+
+
+async def get_db():
+    async with AsyncSessionLocal() as db:
+        yield db
+
+
+@router_temp.get("/", response_model=list[schemas.Temperature])
+async def read_city(db: AsyncSession = Depends(get_db)):
+    return await crud.get_temp(db=db)
+
+
+@router_temp.get("/{city_id}/", response_model=schemas.Temperature)
+async def read_single_city(city_id: int, db: AsyncSession = Depends(get_db)):
+    temp = await crud.get_single_temp(db=db, city_id=city_id)
+    if temp is None:
+        raise HTTPException(status_code=404, detail="City not found")
```

Comment:
The error message 'City not found' might be misleading when dealing with temperature records. Consider changing it to 'Temperature record not found' to provide more accurate feedback to the client.